pkg/otlp/collector.go

```diff
@@ -222,14 +230,17 @@ func BuildAndStart(ctx context.Context, cfg config.Config, s serializer.MetricSe
 	return p, nil
 }
 
-func NewPipelineFromAgentConfig(cfg config.Config, s serializer.MetricSerializer) (*Pipeline, error) {
+func NewPipelineFromAgentConfig(cfg config.Config, s serializer.MetricSerializer, logsAgentChannel chan *message.Message) (*Pipeline, error) {
 	pcfg, err := FromAgentConfig(cfg)
 	if err != nil {
 		pipelineError.Store(fmt.Errorf("config error: %w", err))
 		return nil, pipelineError.Load()
 	}
+	if err := checkAndUpdateCfg(cfg, &pcfg, logsAgentChannel); err != nil {
```

Comment:
Why are we failing to start the collector if logs aren't enabled? Shouldn't we start the other receivers for traces and metrics?